pliers/filters/text.py

```diff
@@ -22,52 +23,78 @@ class TextFilter(Filter):
+            stemming or lemmatizing.
         args, kwargs: Optional positional and keyword args passed onto the
-            nltk stemmer.
+            nltk stemmer/lemmatizer.
     '''
 
-    stemmers = {
+    _stemmers = {
         'porter': 'PorterStemmer',
         'snowball': 'SnowballStemmer',
         'lancaster': 'LancasterStemmer',
         'isri': 'ISRIStemmer',
         'regexp': 'RegexpStemmer',
-        'rslp': 'RSLPStemmer'
+        'rslp': 'RSLPStemmer',
+        'wordnet': 'WordNetLemmatizer'
     }
 
-    _log_attributes = ('stemmer', 'tokenize')
-
-    def __init__(self, stemmer='porter', tokenize=True, *args, **kwargs):
+    _log_attributes = ('stemmer', 'tokenize', 'case_sensitive')
 
+    @requires_nltk_corpus
+    def __init__(self, stemmer='porter', tokenize=True, case_sensitive=False,
+                 *args, **kwargs):
         if isinstance(stemmer, string_types):
-            if stemmer not in self.stemmers:
-                valid = list(self.stemmers.keys())
+            if stemmer not in self._stemmers:
+                valid = list(self._stemmers.keys())
                 raise ValueError("Invalid stemmer '%s'; please use one of %s."
                                  % (stemmer, valid))
-            stemmer = getattr(stem, self.stemmers[stemmer])(*args, **kwargs)
+            stemmer = getattr(stem, self._stemmers[stemmer])(*args, **kwargs)
         elif not isinstance(stemmer, (stem.StemmerI, stem.WordNetLemmatizer)):
             raise ValueError("stemmer must be either a valid string, or an "
                              "instance of class StemmerI.")
         self.stemmer = stemmer
         self.tokenize = tokenize
+        self.case_sensitive = case_sensitive
         super(WordStemmingFilter, self).__init__()
 
+    @requires_nltk_corpus
     def _filter(self, stim):
+        pos_map = {
+            'ADJ': 'a',
+            'ADJ_SAT': 's',
+            'ADV': 'r',
+            'NOUN': 'n',
+            'VERB': 'v'
+        }
+
+        def pos_wordnet(txt):
+            pos_tagged = dict(nltk.pos_tag(txt, tagset='universal'))
+            pos_tagged = {t: pos_map[tag] if tag in pos_map else 'n'
+                          for t, tag in pos_tagged.items()}
+            return pos_tagged
+
+        tokens = [stim.text]
         if self.tokenize:
-            tokens = stim.text.split()
-            stemmed = ' '.join([self.stemmer.stem(tok) for tok in tokens])
+            tokens = nltk.word_tokenize(tokens[0])
+        tokens = [t if self.case_sensitive else t.lower() for t in tokens]
+        if not isinstance(self.stemmer, stem.WordNetLemmatizer):
+            stemmed = ' '.join([self.stemmer.stem(t) for t in tokens])
         else:
-            stemmed = self.stemmer.stem(stim.text)
+            pos_tagged = pos_wordnet(tokens)
+            stemmed = ' '.join([self.stemmer.lemmatize(t, pos=pos_tagged[t])
+                                for t in tokens])
         return TextStim(stim.filename, stemmed)
```

Comment:
We probably want to make sure we pass through all of the other fields in the `stim`, so either pass them all explicitly (e.g., `stim.onset`), or just `copy.copy` the stim object and overwrite the text field (the former is probably better).